nexus/pkg/filestream/loop_transmit.go

```diff
@@ -0,0 +1,161 @@
+package filestream
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+var completeTrue bool = true
+
+type chunkData struct {
+	fileName string
+	fileData *chunkLine
+	Exitcode *int32
+	Complete *bool
+}
+
+type chunkLine struct {
+	chunkType ChunkFile
+	line      string
+}
```

Comment:
maybe re-think these types, it was quite awkward in the send function when we combining the chunks

Reply:
yes, at least they are more private now and tied to the "transmit". but this was getting more attention in the PR which was going to make sending different types of data better.